### PR TITLE
[FIX] website_sale: improve internal ref search

### PR DESCRIPTION
**Pull request by myhy-odoo**
Problem:
Results do not show when users search the internal reference of the product template on the ecommerce page if the product has other variants aside from itself. The search only searches internal reference of product variants and not the product template.

Steps to reproduce on Runbot 15:
- Install Sales and eCommerce
- Create a product (can be sold), Set its Internal reference, and Add attributes and variants to the product --> Save Note: the internal ref of the product variants are not set by default
- Go to ecommerce webpage and search the internal reference --> no product found

Proposed solution:
 Add a condition to the domain to also search based on the internal reference of the product template

opw-3212883

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by myhy-odoo** [comment]
Hello @luve-odoo ,

Can you help review this PR to check if the fix is satisfactory or not? Also, what is the usual expected time to receive a response for PRs? Thank you for your help in advance!

Best,
Mylyna Hy

**Review by AntoineVDV — APPROVED** [review]
Hello, thanks for the fix.

I'm not familiar with this mixin, but it seems good to me, and it does fix the bug :+1: 

I just need to confirm with eCommerce's PO, @paju-odoo, that we do want to enable customers to search the product template's internal reference when it has variants. This is unclear to me because, [when the product template has variants, the internal reference is hidden from the form view](https://github.com/odoo/odoo/blob/15.0/addons/product/views/product_template_views.xml#L44).

**Comment by AntoineVDV** [comment]
> Also, what is the usual expected time to receive a response for PRs? Thank you for your help in advance!

This varies from team to team, but it's generally safe to proceed with the merge of the PR when you receive zero answers in 1-2 weeks. Maybe ask on the dedicated Discord channel (`rd-fun-sales` in this case) after a week.

In the case of the eCommerce app, we are proceeding with the hand-over from @odoo/rd-ecommerce-documents to @odoo/rd-ecommerce, so you may tag the latter on PRs that are waiting for a review from the former team ;)

**Comment by AntoineVDV** [comment]
Green light :green_circle: 

@robodoo r+

**Comment by AntoineVDV** [comment]
@robodoo r-

(re-consideration in progress)

**Comment by AntoineVDV** [comment]
@robodoo r+ :rocket:

**Comment by myhy-odoo** [comment]
Hello @AntoineVDV ,

Thank you for the review and update!

**Pull request closed without merging**
